### PR TITLE
Use letters to de-dupe imported questions

### PR DESCRIPTION
**Pull request by avaleske**
### Description

Use letters instead of numbers to de-dupe imported questions during program migration.

Details:
- Numbers are stripped from question names when generating keys[1], so they end up colliding when exported or used to access ApplicantData.
- Instead, we use "excel column name" style letters to de-duplicate questions with conflicting names.

[1] https://github.com/civiform/civiform/blob/0c57746d98eaba7471d65bd8fe50d792701c2990/server/app/services/question/types/QuestionDefinition.java#L60

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [x] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [x] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`

### Instructions for manual testing

- Try importing questions with duplicate names
- Also try importing a program with questions that both need to be deduplicated, to ensure they don't collide with each other.

### Issue(s) this completes

Part of #9212

### CODE REVIEW

**Comment by avaleske** [comment]
* **#9240** <a href="https://app.graphite.dev/github/pr/civiform/civiform/9240?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/civiform/civiform/9240?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: server/app/services/migration/ProgramMigrationService.java [hunk 1]
```diff
@@ -13,22 +13,33 @@
...
 
 /**
  * A service responsible for helping admins migrate program definitions between different
  * environments.
  */
 public final class ProgramMigrationService {
+  private static final String CONFLICTING_QUESTION_FORMAT =
+      "%s - %s"; // this will transform to a key formatted like `%s__%s`
+  private static final Pattern SUFFIX_PATTERN = Pattern.compile(" - [a-z]+$");
+  private static final String ALPHABET = "abcdefghijklmnopqrstuvwzyz";
```
Comment (gwendolyngoetz) [inline server/app/services/migration/ProgramMigrationService.java:41]:
```suggestion
  private static final String ALPHABET = "abcdefghijklmnopqrstuvwxyz";
```

Missing `x` and have `z` twice

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

**Review by gwendolyngoetz — APPROVED** [review]

File: server/app/services/migration/ProgramMigrationService.java [hunk 2]
```diff
@@ -84,18 +95,21 @@ public ErrorAnd<ProgramMigrationWrapper, String> deserialize(String programJson)
 
   /**
    * Checks if there are existing questions that match the admin id of any of the incoming
-   * questions. If a match is found, generate a new admin name of the format "orginal admin name-n".
+   * questions. If a match is found, generate a new admin name of the format "orginal admin name -
+   * a".
```
Comment (dkatzz) [inline server/app/services/migration/ProgramMigrationService.java:99]: **nitpick:** Can you put these on the same line? maybe move down the whole admin name

File: server/app/services/migration/ProgramMigrationService.java [hunk 3]
```diff
@@ -105,32 +119,81 @@ public ImmutableMap<String, QuestionDefinition> maybeOverwriteQuestionName(
...
+  String findUniqueAdminName(String adminName, List<String> newNamesSoFar) {
+    if (!nameHasConflict(adminName, newNamesSoFar)) {
+      return adminName;
+    }
+
+    // If the question name contains a suffix of the form " - a" (for example "admin name - a"), we
+    // want
+    // to strip off the " - n" to find the base name. This also allows us to correctly increment the
+    // suffix of the base admin name so we don't end up with admin names like "admin name - a - a".
+    Matcher matcher = SUFFIX_PATTERN.matcher(adminName);
```
Comment (dkatzz) [inline server/app/services/migration/ProgramMigrationService.java:136]: **question:** Is it possible for something like "Date of birth - parent" or something to get matched? Could we add a test for that use case?

**Review by rockycodes — COMMENTED** [review]
I took a quick look and it makes sense/looks good to me! I can look in more depth on Monday, but no need to wait on me to merge unless you want to.

File: server/app/services/migration/ProgramMigrationService.java (hunk 3 above)
Comment (avaleske) [reply server/app/services/migration/ProgramMigrationService.java:139]: Discussed in slack, updated to use ` -_- ` as the delimiter.

File: browser-test/src/admin/admin_program_migration.test.ts [hunk 4]
```diff
@@ -390,9 +390,9 @@ test.describe('program migration', () => {
       await expect(programDataDiv).toContainText('What is your address?')
       // question help text
       await expect(programDataDiv).toContainText('help text')
-      // admin name (should be updated with "-1" on the end)
+      // admin name (should be updated with " - a" on the end)
       await expect(programDataDiv).toContainText(
-        'Admin name: Sample Address Question-1',
+        'Admin name: Sample Address Question - a',
```
Comment (dkatzz) [inline browser-test/src/admin/admin_program_migration.test.ts:395]: **suggestion:** This and related comments to be updated too!

**Review by dkatzz — APPROVED** [review]

**Review by dkatzz — APPROVED** [review]

**Pull request merged**